### ISSUE TITLE
Espejo does not ignore everything defined in ignoreNames array

### ISSUE THREAD

**Issue by ZuGao** [vshn/espejo#107]
## Describe the bug

Some regex entries in _ignoreNames_ list are not being enforced.

## Additional context

Espejo does not ignore "csl-*" namespaces in an OpenShift cluster with the following ignoreNames list:
```yml
namespaceSelector:
  ignoreNames:
  - openshift
  - openshift-*
  - syn
  - default
  - '*-production'
  - '*-prod'
  - csl-*
  - kube-system
```

P.S. For security reasons some words have been changed in the above SyncConfig
## Logs

No relevant logs for this issue has been found.

## Expected behavior

Espejo should not create LimitRanges for "csl-*" namespaces.

## To Reproduce

1. Define several namespaces with names that are under these regex definitions:
*   - openshift
    - openshift-*
    - syn
    - default
    - '*-production'
    - '*-prod'
    - csl-*
    - kube-system
2. oc -n \<espejo-namespaces\> apply -f limitranges.yaml
where limitranges.yaml:
```yml
apiVersion: sync.appuio.ch/v1alpha1
kind: SyncConfig
metadata:
  name: limitrange
  namespace: <espejo-namespaces>
spec:
  forceRecreate: true
  namespaceSelector:
    ignoreNames:
    - openshift
    - openshift-*
    - syn
    - default
    - '*-production'
    - '*-prod'
    - csl-*
    - kube-system
    labelSelector:
      matchExpressions:
      - key: name
        operator: Exists
  syncItems:
  - apiVersion: v1
    kind: LimitRange
    metadata:
      name: default-limits
    spec:
      limits:
      - default:
          cpu: 500m
          memory: 128Mi
        defaultRequest:
          cpu: 10m
          memory: 32Mi
        type: Container
```

## Environment (please complete the following information):

- Image Version: espejo:v0.3.2
- K8s Version: v1.11.0+d4cacc0
- K8s Distribution: OpenShift

**Comment by ccremer** [comment]
If you delete the limitrange object from the `csl-*` namespaces, do they get recreated?
Bear in mind, that if the csl-* entry got added later in the list, Espejo does not delete the generated resources unless it's explicitly in the `deleteItems` spec.

**Comment by ZuGao** [comment]
> If you delete the limitrange object from the `csl-*` namespaces, do they get recreated? Bear in mind, that if the csl-* entry got added later in the list, Espejo does not delete the generated resources unless it's explicitly in the `deleteItems` spec.

* The LimitRange objects would get recreated every time.
* The problem is that they should have never been created in the first place.

Let's suppose that I created those LimitRanges by mistake with a previous SyncConfig manifest which did not have `csl-*` regex and later I had added it. If I manually delete LimitRanges from `csl-*` namespaces , I expect that Espejo does not recreate them again as the new `csl-*` value from `ignoreNamespaces` list should be enforced.

**Comment by ccremer** [comment]
One thing I noticed is that you are running an older Espejo version. Please upgrade to v0.4.0. While it's unlikely that v0.4 fixes your issue, If there's any bug it will be fixed in the v0.4 going forward, not v0.3.x, just FYI.

Also, please do share the logs of a reconcile run (redacted is also fine), even if it's seemingly "irrelevant". I first need to understand the problem before being able to fix it :)

**Comment by ccremer** [comment]
Closing issue since that is a misconfiguration on user's end.
